File: java/src/main/java/model/PaperSlip.java
package model;

public class PaperSlip {

    private final String person;
    private final String neighbor;

    public PaperSlip(String person, String neighbor) {
        this.person = person;
        this.neighbor = neighbor;
    }

    public static PaperSlip parse(String s) {
        String[] split = s.split(",");
        return new PaperSlip(split[0], split[1]);
    }

    public String getPerson() {
        return person;
    }

    public String getNeighbor() {
        return neighbor;
    }
}

File: CSharp/CSharp/Model/PaperSlip.cs
namespace CSharp.Model
{
    public class PaperSlip
    {
        
        public string Person { get; }
        public string Neighbour { get; }
        
        public PaperSlip(string Person, string Neighbour)
        {
            this.Person = Person;
            this.Neighbour = Neighbour;
        }

        public static PaperSlip Parse(string line)
        {
            string[] split = line.Split(',');
            return new PaperSlip(split[0], split[1]);
        }
    }
}